include/topic_tools/tool_base_node.hpp

```diff
@@ -29,25 +29,24 @@ class ToolBaseNode : public rclcpp::Node
 public:
   ToolBaseNode(const std::string & node_name, const rclcpp::NodeOptions & options);
 
-private:
+protected:
+  virtual void initialize() = 0;
```

Comment:
could you add a docstring for this interface describing its purpose?

Reply:
the reason that i ask is that this does not seem to be used polymorphically at all - instead it is an internal implementation detail of derived classes (logic which could go in the constructor instead, since it is called from there). Wondering if it is actually an appropriate virtual interface at all.

Reply:
moved initialize() into constructor at https://github.com/ros-tooling/topic_tools/pull/26/commits/39e7ef290cb515a6e5fae64455136ad723eb3f9f